Source/monster.cpp

```diff
@@ -4758,69 +4758,43 @@ bool LineClearMissile(Point startPoint, Point endPoint)
 
 bool LineClear(bool (*Clear)(int, Point), int entity, Point startPoint, Point endPoint)
 {
-	int d;
-	int xincD, yincD, dincD, dincH;
-	bool done = false;
-
 	Point position = startPoint;
 
-	int dx = endPoint.x - position.x;
-	int dy = endPoint.y - position.y;
-	if (abs(dx) > abs(dy)) {
-		if (dx < 0) {
-			std::swap(position, endPoint);
-			dx = -dx;
-			dy = -dy;
-		}
-		if (dy > 0) {
-			d = 2 * dy - dx;
-			dincD = 2 * dy;
-			dincH = 2 * (dy - dx);
-			yincD = 1;
+	int d1 = endPoint.x - position.x;
+	int d2 = endPoint.y - position.y;
+
+	bool horizontal = abs(d1) > abs(d2);
+
+	if (!horizontal)
+		std::swap(d1, d2);
+
+	if (d1 < 0) {
+		std::swap(position, endPoint);
+		d1 = -d1;
+		d2 = -d2;
+	}
+
+	int incD = (d2 > 0) ? 1 : -1;
+	int d = 2 * d2 - (d1 * incD);
+	int dincD = 2 * d2;
+	int dincH = 2 * (d2 - (d1 * incD));

```

Comment:
These variable names are very poor IMHO. Can we rename them to be more specific?

Reply:
Do you have any suggestions?

Reply:
I don't know what the method does and I don't know what the poor names mean, so no, not for now. I was hoping you would know of better names since you seem to know what's going on here.

Reply:
If I could I would have. The internals is still a bit cryptic, but at least now it's not repeated 4 times.